folly/setup.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+
+# Do not call directly, use cmake
+#
+# Cython requires source files in a specific structure, the structure is
+# created as tree of links to the real source files.
+
+from setuptools import setup, Extension
+from Cython.Build import cythonize
+from Cython.Compiler import Options
+import os
+
+Options.fast_fail = True
+
+ext = Extension("folly.executor",
+                sources=['folly/executor.pyx'],
+   include_dirs=[os.getcwd()+"/..", os.getcwd()],
+   libraries=['folly', 'glog', 'double-conversion', 'iberty'])
```

Comment:
I'm assuming the dependencies on `glog`, `double-conversion` and `iberty` are only indirect dependencies through `libfolly`.

It would be nicer to pass this information in to the setup module, as the list of dependencies can be somewhat dynamic based on what was detected by CMake at folly's configure time.
In the CMake code, `get_target_property(YOUR_VAR_NAME folly INTERFACE_LINK_LIBRARIES)` will get the list of libraries that folly depends on.

Reply:
I shall experiment with it and see what I can do.

Reply:
I've spent quite a while looking into this: I cannot see an elegant way to pass the values form `get_target_property(YOUR_VAR_NAME folly INTERFACE_LINK_LIBRARIES)` through to setup.py. To give an example the output of:

```
Threads::Threads;/usr/lib/x86_64-linux-gnu/libboost_context.so;/usr/lib/x86_64-linux-gnu/libboost_chrono.so;/usr/lib/x86_64-linux-gnu/libboost_date_time.so;/usr/lib/x86_64-linux-gnu/libboost_filesystem.so;/usr/lib/x86_64-linux-gnu/libboost_program_options.so;/usr/lib/x86_64-linux-gnu/libboost_regex.so;/usr/lib/x86_64-linux-gnu/libboost_system.so;/usr/lib/x86_64-linux-gnu/libboost_thread.so;/usr/lib/x86_64-linux-gnu/libboost_atomic.so;/usr/lib/x86_64-linux-gnu/libpthread.so;/usr/lib/x86_64-linux-gnu/libdouble-conversion.so;/usr/lib/x86_64-linux-gnu/libglog.so;/usr/lib/x86_64-linux-gnu/libevent.so;/usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so;/usr/lib/x86_64-linux-gnu/libz.so;/usr/lib/x86_64-linux-gnu/libbz2.so;/usr/lib/x86_64-linux-gnu/liblzma.so;/usr/lib/x86_64-linux-gnu/liblz4.so;/usr/lib/x86_64-linux-gnu/libzstd.so;/usr/lib/x86_64-linux-gnu/libsnappy.so;/usr/lib/x86_64-linux-gnu/libiberty.a;dl;gflags
```

In fact its probably easier to read the libfolly.pc (pkg-config) file back and use that from within setup.py.
This rubs against one of the most challenging parts of building with two systems, CMake and Cython's distutils.

Reply:
Getting the info from libfolly.pc would be nice if that isn't too hard.

Landing the current code as-is for now is also okay with me, and we can come back later to improve it if we need to.   One thing I might check is if libelf and libdwarf need to be included here if they were found at configure time.  If they are found FOLLY_USE_SYMBOLIZER will default to ON, and I think folly's init code may then depend on these libraries.